scripts/entrypoint.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+## Inspired by https://github.com/kartoza/docker-geoserver
+set -e
+
+## install GeoServer extensions before starting the tomcat
+echo "Starting installation of extensions"
+/scripts/install-extensions.sh
+echo "\nFinished installation of extensions"
+
+export GEOSERVER_OPTS="-Djava.awt.headless=true -server \
+       -Dfile.encoding=UTF8 \
+       -Djavax.servlet.request.encoding=UTF-8 \
+       -Djavax.servlet.response.encoding=UTF-8 \
+       -Xms${INITIAL_MEMORY} -Xmx${MAXIMUM_MEMORY} \
+       -XX:SoftRefLRUPolicyMSPerMB=36000 -XX:+UseG1GC \
+       -XX:MaxGCPauseMillis=200 -XX:ParallelGCThreads=20 -XX:ConcGCThreads=5 \
+       -Xbootclasspath/a:${CATALINA_HOME}/lib/marlin-${MARLIN_VERSION}-Unsafe.jar \
+       -Dsun.java2d.renderer=org.marlin.pisces.MarlinRenderingEngine \
+       -Dorg.geotools.coverage.jaiext.enabled=${JAIEXT_ENABLED}"
```

Comment:
some of these options may not be valid for Java 11 (don't know which off the top of my head)

Reply:
I tend to run with Java 8 and settings very similar to what you have here.

The default for memory is 1/4 system memory, not sure what the docker image ends up?

The `UseG1GC` is the default for Java 11, also marlin is baked in if using Java 11.

The number of threads to devote to garbage collection is is a tradeoff based on the number of threads available.

Reply:
> not sure what the docker image ends up

it depends if the container is created with resource limits like [`--memory`](https://docs.docker.com/config/containers/resource_constraints/#limit-a-containers-access-to-memory) if the container tries to allocate more than allowed it will be killed, but users can specify those using env variables 

https://github.com/geoserver/docker/blob/eee16c94e4a4fafdcb8b85b08f1583cb0efa6dbd/Dockerfile#L20-L21

https://github.com/geoserver/docker/blob/eee16c94e4a4fafdcb8b85b08f1583cb0efa6dbd/scripts/entrypoint.sh#L14

Reply:
I switched to OpenJDK 11 in a03ee5a which also adjusts the lines above